core/src/type_info.rs

```diff
@@ -31,7 +31,8 @@ const LEGACY_DATE: BsonTypeInfo = BsonTypeInfo {
 
 // order of array is by SqlDataType, since that is the ordering of the
 // SQLGetTypeInfo result set according to the spec
-const DATA_TYPES: [BsonTypeInfo; 22] = [
+const DATA_TYPES: [BsonTypeInfo; 23] = [
+    BsonTypeInfo::VARCHAR,             // SqlDataType(12)
```

Comment:
this should be in order of sql data type, so I think this needs to come after `LEGACY_DATE`